Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/SyncKronosToShiftsController.cs

```diff
@@ -98,57 +98,119 @@ public ActionResult SyncKronosToShifts(string isRequestFromLogicApp)
         }
 
         /// <summary>
-        /// Process all the entities from kronos to shifts.
+        /// Process all the entities from Kronos to Shifts.
         /// </summary>
         /// <param name="isRequestFromLogicApp">true if the call is coming from logic app, false otherwise.</param>
         /// <returns>Returns task.</returns>
+#pragma warning disable CA1031 // Do not catch general exception types - Suppressing as there may be unknown points of failure.
         private async Task ProcessKronosToShiftsShiftsAsync(string isRequestFromLogicApp)
         {
+            var isOpenShiftRequestSyncSuccessful = false;
+            var isSwapShiftRequestSyncSuccessful = false;
+            var isMapPayCodeTimeOffReasonsSuccessful = false;
+
             try
             {
                 this.telemetryClient.TrackTrace($"{Resource.ProcessKronosToShiftsShiftsAsync} start at: {DateTime.UtcNow.ToString("O", CultureInfo.InvariantCulture)}" + " for isRequestFromLogicApp: " + isRequestFromLogicApp);
 
-                // Sync open shifts from kronos to shifts.
+                // Sync open shifts from Kronos to Shifts.
                 await this.openShiftController.ProcessOpenShiftsAsync(isRequestFromLogicApp).ConfigureAwait(false);
 
                 this.telemetryClient.TrackTrace($"{Resource.ProcessOpenShiftsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            }
+            catch (Exception ex)
+            {
+                this.telemetryClient.TrackException(ex);
```

Comment:
You need to log a message here as well, at the Error severity level

Reply:
Here and in the equivalent catch blocks.

Reply:
@aosolis - Ack'd sir! I have made sure to have the following:
```cs
this.telemetryClient.TrackTrace($"An error happened when syncing timeoff requests: {ex.Message}", SeverityLevel.Error);
```
Please confirm if that is what you mean